pontoon/contributors/static/js/settings.js

```diff
@@ -22,20 +21,29 @@ $(function () {
     const originalValue = this.dataset.originalValue || '';
 
     if (value === originalValue) {
+      self.parents('.field').find('.errorlist').empty();
+
+      if (attribute === 'contact_email') {
+        if (value !== '') {
+          self.parents('.field').find('.help').addClass('hide');
+          self.parents('.field').find('.verify').removeClass('hide');
+        } else {
+          self.parents('.field').find('.help').removeClass('hide');
+          self.parents('.field').find('.verify').addClass('hide');
+        }
+      }
       return;
     }
 
-    this.dataset.originalValue = value;
-
```

Comment:
You had the right idea here, this should still happen before the API call is made. Remember that it may take time for the request to go to the server and come back, which is when the `success()` handler will be called. During that time, having the `dataset.originalValue` set will allow duplicate requests to be avoided.

The desired change from the previous is that in the `error()` handler you should include something like:
```js
this.dataset.originalValue = originalValue;
```